From e9fd0d807d339a687d3a3f4aba03585be5f3e345 Mon Sep 17 00:00:00 2001
From: Giovanni Santini <giovannisantini93@yahoo.it>
Date: Tue, 10 Feb 2015 10:18:47 +0100
Subject: [PATCH] Set up Jenkins build variables

---
 tools/buildinfo.sh | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/tools/buildinfo.sh b/tools/buildinfo.sh
index 7850611..9b4f6f0 100755
--- a/tools/buildinfo.sh
+++ b/tools/buildinfo.sh
@@ -15,8 +15,16 @@ if [ -z "$BUILD_UTC_DATE" ] ; then
 fi
 echo "ro.build.date.utc=$BUILD_UTC_DATE"
 echo "ro.build.type=$TARGET_BUILD_TYPE"
-echo "ro.build.user=$USER"
-echo "ro.build.host=`hostname`"
+
+if [ -z "$BUILD_USER_ID" ]
+then
+	echo "ro.build.user=$USER"
+	echo "ro.build.host=`hostname`"
+else
+	echo "ro.build.user=$BUILD_USER_ID"
+	echo "ro.build.host=moltenmotherboard"
+fi
+
 echo "ro.build.tags=$BUILD_VERSION_TAGS"
 echo "ro.product.model=$PRODUCT_MODEL"
 echo "ro.product.brand=$PRODUCT_BRAND"
-- 
2.3.0

